### DISCUSSION TITLE
Import folders into albums

### DISCUSSION THREAD

**Discussion by pavel-lukes** [photoprism/photoprism#3684] [Q&A]
I would like to initiate the import of dozens of directories and immediately create albums from them. What's the best approach? Is there any form of data structure that would facilitate this?

**Comment by graciousgrey** [comment]
Do you want to use the [import or indexing](https://docs.photoprism.app/user-guide/library/) method? 

In case you index your files you can find folder-albums for each folder of your originals directory in the [folder section](https://docs.photoprism.app/user-guide/organize/folders/). The folder albums are created automatically and are updated each time a photo is added to the folder.

If you want to use the import method you could use the [web upload](https://docs.photoprism.app/user-guide/library/upload/) and select a album in the upload dialog. Alternatively after importing files you could search for the original folder name by using the [search filter "original"](https://docs.photoprism.app/user-guide/search/filters/#filter-reference). You can then [add the search results to an album](https://docs.photoprism.app/user-guide/organize/albums/#add-photos-to-album).

**Reply by nickelsound** [reply]
Hi, I would say that the best method for me is 'indexing,' but can I automatically transform the 'folders' into albums? I'm talking about thousands of folders.

**Reply by graciousgrey** [reply]
There is no way to do this automatically at this time. You can do it manually as described [here](https://docs.photoprism.app/user-guide/organize/folders/#create-albums-from-folders). However folders albums are very similar to manually managed albums - so depending on your use case you might not need to create manually managed albums from them.
